framework/include/interfaces/Coupleable.h

```diff
@@ -240,6 +240,62 @@ class Coupleable
   std::vector<const ADVectorVariableValue *>
   adCoupledVectorValues(const std::string & var_name) const;
 
+  /**
+   * Returns value of a coupled array variable for a given tag
+   * @param var_name Name of coupled array variable
+   * @param tag vector tag ID
+   * @param comp Component number for vector of coupled variables
+   * @return Reference to a VariableValue for the coupled array variable
+   * @see Kernel::_u
+   */
+  virtual const ArrayVariableValue &
```

Comment:
how is this an arrayvariablevalue if you are taking a component

Reply:
Same as `coupledArrayValue` below, here, in case of providing the list of array variable names as the input argument, the component refers to the `i-th` array variable (not the `i-th` component of array variable). Thus, the return type should  be `ArrayVariablesValue`. 

```c++
const ArrayVariableValue &
Coupleable::coupledArrayValue(const std::string & var_name, unsigned int comp) const
```

Reply:
this is really confusing.
- use a different name for component if it s a different indexing. 
- add 's' to the relevant inputs of the functions (var_nameS)

why would you need to do this though? Cant you call coupledValue with the right variable name instead of passing a vector of names and having all this machinery to extract one of the names?

Reply:
I agree this is really confusing.
Due to the code consistencies with other functions such as `coupledArrayValue` and `coupledVectorTagValue`, I implemented new function such way.

@lindsayad 
Do you have any suggestion on this issue?

Reply:
these issues are pre-existing your PR I see that.

Reply:
Alex is on PL for a few weeks so we ll resolve this on our own

I think having better docstrings, better names for routine arguments and possibly more templating will be enough

Reply:
let's leave the templating for later. Logan says it s complicated.

Reply:
Actually, I am trying on it and there are some errors/issues beyond my C++ knowledge ;) 
Good to know it is complicated. I will leave that task for later.